hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableECContainerProvider.java

```diff
@@ -159,13 +160,21 @@ public ContainerInfo getContainer(final long size,
     // If we get here, all the pipelines we tried were no good. So try to
     // allocate a new one.
     try {
-      synchronized (this) {
-        if (openPipelineCount < maximumPipelines) {
+      if (openPipelineCount >= maximumPipelines) {
```

Comment:
Should this exclude DECOMMISSIONED state nodes?

What if more nodes take to offline mode? then this count puts the pressure on cluster with more openpipelines? ( May be rare case as people may not take many nodes at a time though )

Reply:
Thanks, updated to only consider in-service healthy nodes.